### PR TITLE
Merge master to future

### PR DESCRIPTION
**Pull request by nguerrera**

### CODE REVIEW

**Comment by nguerrera** [comment]
@ramarag @eerhardt Ideas on why tests are failing on Linux only;

```
/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: The "ResolvePublishAssemblies" task failed unexpectedly. [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 /mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary. [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 /mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: at System.ThrowHelper.ThrowKeyNotFoundException() [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 /mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: at System.Collections.Generic.Dictionary`2.get_Item(TKey key) [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 /mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: at Microsoft.NET.Build.Tasks.ProjectContext.GetIntersection(IDictionary`2 collection1, IDictionary`2 collection2) [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 /mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: at Microsoft.NET.Build.Tasks.ProjectContext.GetRuntimeLibraries(IEnumerable`1 privateAssetPackageIds) [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 /mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: at Microsoft.NET.Build.Tasks.PublishAssembliesResolver.Resolve(ProjectContext projectContext) [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 /mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: at Microsoft.NET.Build.Tasks.ResolvePublishAssemblies.ExecuteCore() [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 /mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: at Microsoft.NET.Build.Tasks.TaskBase.Execute() [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 /mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 /mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets(233,5): error MSB4018: at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext() [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/bin/Debug/Tests/It_publishes_projects_with_filter_and_rid/SimpleDependencies/SimpleDependencies.csproj] [/mnt/j/workspace/dotnet_sdk/future/debug_ubuntu14.04_prtest/build/build.proj]
15:13:31 
```

**Comment by eerhardt** [comment]
It's failing in a new method @ramarag added  `Microsoft.NET.Build.Tasks.ProjectContext.GetIntersection`.

@ramarag - can you take a look and push a fix to this branch?

**Comment by ramarag** [comment]
looking now

**Comment by ramarag** [comment]
@nguerrera @eerhardt i have in pushed the fix

File: src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs [hunk 1]
```diff
@@ -196,10 +196,10 @@ public IEnumerable<string> GetTopLevelDependencies()
             }
             foreach (var entry in iterated)
             {
-                LockFileTargetLibrary library = lookup[entry.Key];
 
-                if (library != null)
+                if (lookup.ContainsKey(entry.Key))
```
Comment (eerhardt) [inline src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs]: @ramarag - Please use `TryGetValue`.

Comment (ramarag) [inline src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs]: done

**Comment by ramarag** [comment]
@dotnet-bot test this please

**Comment by eerhardt** [comment]
Sometimes closing and reopening PRs will kick @dotnet-bot to run....

**Pull request merged**